Improve docs on how to register a custom operator for `@whereConditions` (#1156)
nuwave/lighthouse: docs/master/eloquent/complex-where-conditions.md

You may register a custom `\Nuwave\Lighthouse\WhereConditions\Operator` through a service provider.
This may be necessary if your database uses different SQL operators then Lighthouse's default or you
want to extend/restrict the allowed operators.

First create a class that implements `\Nuwave\Lighthouse\WhereConditions\Operator`, e.g. `\App\GraphQL\CustomSQLOperator`.
In following example we will extend `\Nuwave\Lighthouse\WhereConditions\SQLOperator` that actually implements 
`\Nuwave\Lighthouse\WhereConditions\Operator` and modify result of `enumDefinition()` to include `ILIKE` operator to the
end of the `SQLOperator` enum. 

```php
use Nuwave\Lighthouse\WhereConditions\SQLOperator;

> class CustomSQLOperator extends SQLOperator

Reviewer: I think this is quite fragile, so i would not recommend doing this to users.

Author: Ye... ok. do you have an example for me? :D

Reviewer: We have been discussing some ways that allow extending the `SQLOperator` more cleanly, let's hold off on our recommended solution until we implemented something better.

Doing string manipulation on the definition string is really hacky.

{
    public function enumDefinition(): string
    {
        $originalDefinition = parent::enumDefinition();
        $additionOperators = '
    "Simple pattern matching (`ILIKE`)"
    ILIKE @enum(value: "ILIKE")
';
        $pos = strrpos($originalDefinition, '}');
        return substr_replace($originalDefinition, $additionOperators, $pos, 0);
    }
}
```

To tell lighthouse to use your custom Operator definition, you have to bind it in one of the service providers that comes 
after `\Nuwave\Lighthouse\WhereConditions\WhereConditionsServiceProvider::class` in your `config/app.php`. E.g. create `\App\Providers\GraphQLServiceProvider`:

```php
use App\GraphQL\CustomSQLOperator;
use Illuminate\Support\ServiceProvider;
use Nuwave\Lighthouse\WhereConditions\Operator;

class GraphQLServiceProvider extends ServiceProvider
{

>     public function register()

Reviewer: ```suggestion
    public function register(): void
```

    {
        $this->app->bind(Operator::class, CustomSQLOperator::class);
    }
}
```

> Don't forget to register you just created service provider in `config/app.php`

Reviewer: ```suggestion
Don't forget to register your new service provider in `config/app.php`:
```
